staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -71,6 +85,8 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (user.Info, bool,
 		}
 	}
 
+	remaining := req.TLS.PeerCertificates[0].NotAfter.Sub(time.Now())
+	clientCertificateExpirationGauge.WithLabelValues(req.TLS.PeerCertificates[0].Subject.CommonName).Set(float64(remaining / time.Second))
```

Comment:
Would you consider the number of nodes in the cluster to be too high cardinality? It would be useful to have the information to identify which nodes are about to drop out of the cluster because they are failing to update certificates.